Source/SwiftLintBuiltInRules/Rules/Lint/NonOptionalStringDataConversionRule.swift

```diff
@@ -0,0 +1,48 @@
+import SwiftSyntax
+
+@SwiftSyntaxRule
+struct NonOptionalStringDataConversionRule: ConfigurationProviderRule {
+    var configuration = SeverityConfiguration<Self>(.warning)
+    static let description = RuleDescription(
+        identifier: "non_optional_string_data_conversion",
+        name: "Non-Optional String <-> Data Conversion",
+        description: "Prefer using UTF8 encoded strings when converting between `String` and `Data`",
+        kind: .lint,
+        nonTriggeringExamples: [
+            Example("Data(\"foo\".utf8)"),
+            Example("String(decoding: data, as: UTF8.self)")
+        ],
+        triggeringExamples: [
+            Example("\"foo\".data(using: .utf8)"),
+            Example("String(data: data, encoding: .utf8)")
+        ]
+    )
+}
+
+private extension NonOptionalStringDataConversionRule {
+    final class Visitor: ViolationsSyntaxVisitor {
+        override func visitPost(_ node: FunctionCallExprSyntax) {
+            print(node)
+            if let expression = node.calledExpression.as(MemberAccessExprSyntax.self),
```

Comment:
The logic looks okay. However, you may want to implement `visitPost(_ node: MemberAccessExprSyntax)` and `visitPost(_ node: DeclReferenceExprSyntax)` and check therein if their parent node is a `FunctionCallExprSyntax`.